hibike/hibike_message.py

```diff
@@ -405,19 +407,21 @@ def parse_bytes(msg_bytes):
     return HibikeMessage(message_id, payload)
 
 
-def blocking_read_generator(serial_conn, stop_event=None):
+def blocking_read_generator(serial_conn, stop_event=threading.Event()):
     """
-    Yield packets from SERIAL_CONN, stopping if STOP_EVENT exists
-    and is set.
+    Yield packets from SERIAL_CONN, stopping if STOP_EVENT is set.
     """
     zero_byte = bytes([0])
     packets_buffer = bytearray()
-    while stop_event is None or not stop_event.is_set():
-
+    # Switch to nonblocking mode so that we don't get stuck reading
+    old_timeout = serial_conn.timeout
+    serial_conn.timeout = 0
```

Comment:
Is there really a point to manipulating the timeout like this? Who else would change it/ when do we ever want a timeout?

Reply:
It's useful if you want to do blocking reads, then `timeout = None`.